_drafts/conviva-dag-process-time-prod-issue-draft.md

```diff
@@ -103,7 +111,7 @@ First, let's look a little more closely at how concurrent hash maps typically op
 ⚠️ Constant inter-core cache line transfers = degraded perf
 ```
 
-The fact that this is a single, shared counter causes significant grief under contention because every CPU has a cache, and every cache requires loading data in from RAM. Each core is going to attempt to increment/decrement the same counter, and each modification causes cache invalidation because [cache coherence](https://en.wikipedia.org/wiki/Cache_coherence) dictates that any change to a cache line requires re-fetching from main memory. To understand this better, look at this section below from a great PDF titled [What every systems programmer should know about concurrency](https://assets.bitbashing.io/papers/concurrency-primer.pdf) by [Matt Kline](https://github.com/mrkline).
+In cases where the data is guarded by a single, shared counter contention can arise under high loads. This is because every CPU core attempting to increment or decrement the counter causes cache invalidation due to [cache coherence](https://en.wikipedia.org/wiki/Cache_coherence). Each modification forces the cache line containing the counter to "ping-pong" between cores, leading to degraded performance. To understand this better, look at this section below from a great PDF titled [What every systems programmer should know about concurrency](https://assets.bitbashing.io/papers/concurrency-primer.pdf) by [Matt Kline](https://github.com/mrkline).
```

Comment:
IT doesn't even need to be a single one, the sharded one can also lead to contention if that shard is heavily contended.

Reply:
Yeah, changed it to the following phrasing

"In cases where the data is guarded by a single, shared counter or resides on the same shard"